app/full_text_search.py

```diff
@@ -185,21 +186,36 @@ def create_full_name_without_space_fields(romanize_method, given_name, family_na
     romanized_name_list = []
     romanized_given_name = romanize_method(given_name)
     romanized_family_name = romanize_method(family_name)
+    romanize_method_name = romanize_method.__name__
     full_names = create_full_name_without_space(
         romanized_given_name, romanized_family_name)
     if full_names:
         full_name_given_family, full_name_family_given = full_names
         fields.append(appengine_search.TextField(
-            name='no_space_full_name_romanized_by_'+romanize_method.__name__+'_1',
+            name='no_space_full_name_%d_romanized_by_%s_1' % (
+                index, romanize_method_name),
             value=full_name_given_family))
         romanized_name_list.append(full_name_given_family)
         fields.append(appengine_search.TextField(
-            name='no_space_full_name_romanized_by_'+romanize_method.__name__+'_2',
+            name='no_space_full_name_%d_romanized_by_%s_2' % (
+                index, romanize_method_name),
             value=full_name_family_given))
         romanized_name_list.append(full_name_family_given)
     return fields, romanized_name_list
 
 
+def split_full_name_without_space(romanize_method, full_name):
+    names = []
+    for index in xrange(len(full_name)):
+        family_name = full_name[:index]
+        given_name = full_name[index:]
+        romanized_family_name = romanize_method(family_name)
```

Comment:
Please merge with the master branch first. Then this romanize_method now returns an array instead, so please support it.